backend/src/api/groups/groups.api.ts

```diff
@@ -82,8 +82,8 @@ const initGroupsApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
       const isDeleted = await groupService.delete(Number(id));
 
       return rep
-        .status(isDeleted ? HttpCode.NO_CONTENT : HttpCode.NOT_FOUND)
-        .send();
+        .status(isDeleted ? HttpCode.OK : HttpCode.NOT_FOUND)
```

Comment:
If we don't send deleted entities back, then the status should be 204
```suggestion
        .status(isDeleted ? HttpCode.NO_CONTENT : HttpCode.NOT_FOUND)
```

Reply:
Is it a bad idea to send back 'true' or 'false' and 200 status instead of 204 and empty res?

Reply:
Delete should return 204 if the action has been performed but response doesn't include an entity

Reply:
After I changed the response to 204 I am getting error on front part ‘unexpected end of JSON’. It looks like it is trying to read JSON from a response and obviously it is not there. But I can’t figure out where exactly it is trying to read it and how to fix that ...

Reply:
The reason for it is in `http.service.ts` file of Front-End. Specifically, it's in methods `checkStatus` and `parseJSON` . Here's why:
The Back-End sends no response and status 204. Status 204 means everything is fine, no errors. In that case, `checkStatus` returns `response`. Then `parseJSON` tries to call `response.json()`, but there's no response from the API

Possible solution:
```
private async checkStatus(response: Response): Promise<Response | null> {
    if (!response.ok) {
      const parsedException = await response.json().catch(() => ({
        message: response.statusText,
      }));

      throw new HttpError({
        status: response.status,
        message: parsedException?.message,
      });
    }
   
   if (response.status === 204) {
       return null;
   }

    return response;
  }

  private parseJSON<T>(response: Response | null): Promise<T | null> {
  if (response) {
    return response.json();
    }
    return null
  }
```
Or something similar

Reply:
No, 204 status means everything is OK, so the code inside `if (!response.ok) { ... }` will not run

Reply:
response is OK for all 200-299 status codes 🥲

Reply:
response is OK for all 200-299 status codes

Reply:
Do I get it correct that I would also need to update this method to return null as well as add this option to all the services ? 

  public load<T = unknown>(
    url: string,
    options: Partial<HttpOptions> = {},
  ): Promise<T | null> {
    const {
      method = HttpMethod.GET,
      payload = null,
      contentType,
      hasAuth = true,
      queryString,
    } = options;
    const headers = this.getHeaders(contentType, hasAuth);

    return fetch(this.getUrlWithQueryString(url, queryString), {
      method,
      headers,
      body: payload,
    })
      .then(this.checkStatus)
      .then((res) => this.parseJSON<T>(res))
      .catch(this.throwError);
  }

Reply:
I don't think you need to update any other services. Also, no need to update `load` method. You just can specify the type as `null`, when calling `load`
`this.#http.load<null>(url, opts)`

Reply:
actually we don't want to change http service all

Reply:
To avoid changes in http service you should send isDeleted

Reply:
so switch the return status back to 200?